pkg/sql/opt/optbuilder/mutation_builder.go

```diff
@@ -1393,47 +1393,28 @@ func checkDatumTypeFitsColumnType(col *cat.Column, typ *types.T) {
 func (mb *mutationBuilder) addAssignmentCasts(inScope *scope, outTypes []*types.T) *scope {
 	expr := inScope.expr.(memo.RelExpr)
 
-	// Do a quick check to see if any casts are needed.
-	castRequired := false
-	for i := 0; i < len(inScope.cols); i++ {
-		if !inScope.cols[i].typ.Identical(outTypes[i]) {
-			castRequired = true
-			break
-		}
-	}
-	if !castRequired {
-		// No mutation casts are needed.
-		return inScope
-	}
-
 	projectionScope := inScope.push()
 	projectionScope.cols = make([]scopeColumn, 0, len(inScope.cols))
 	for i := 0; i < len(inScope.cols); i++ {
 		srcType := inScope.cols[i].typ
 		targetType := outTypes[i]
-		if !srcType.Identical(targetType) {
-			// Check if an assignment cast is available from the inScope column
-			// type to the out type.
-			if !tree.ValidCast(srcType, targetType, tree.CastContextAssignment) {
-				ord := mb.tabID.ColumnOrdinal(mb.targetColList[i])
-				colName := string(mb.tab.Column(ord).ColName())
-				err := pgerror.Newf(pgcode.DatatypeMismatch,
-					"value type %s doesn't match type %s of column %q",
-					srcType, targetType, tree.ErrNameString(colName))
-				err = errors.WithHint(err, "you will need to rewrite or cast the expression")
-				panic(err)
-			}
-
-			// Create a new column which casts the input column to the correct
-			// type.
-			variable := mb.b.factory.ConstructVariable(inScope.cols[i].id)
-			cast := mb.b.factory.ConstructAssignmentCast(variable, outTypes[i])
-			mb.b.synthesizeColumn(projectionScope, inScope.cols[i].name, outTypes[i], nil /* expr */, cast)
-		} else {
-			// The column is already the correct type, so add it as a
-			// passthrough column.
-			projectionScope.appendColumn(&inScope.cols[i])
+		// Check if an assignment cast is available from the inScope column
```

Comment:
Does it make sense to add a comment explaining why we are adding assignment casts in all cases, especially in case we change this in the future?